src/client/cl_entity.c

```diff
@@ -42,16 +42,25 @@ static void Cl_ParsePlayerState(const cl_frame_t *delta_frame, cl_frame_t *frame
 /**
  * @return True if the delta is valid and interpolation should be used.
  */
-static _Bool Cl_ValidDeltaEntity(const entity_state_t *from, const entity_state_t *to) {
+static _Bool Cl_ValidDeltaEntity(const cl_frame_t *frame, const cl_entity_t *ent,
+								 const entity_state_t *from, const entity_state_t *to) {
 	vec3_t delta;
 
+	if (frame->delta_frame_num == -1) {
+		return false;
+	}
+
+	if (ent->frame_num != frame->frame_num - 1) {
+		return false;
+	}
+
 	if (from->model1 != to->model1) {
 		return false;
 	}
 
 	VectorSubtract(from->origin, to->origin, delta);
 
-	if (VectorLength(delta) > 256.0) {
+	if (VectorLength(delta) > 64.0) { // MAX_SPEED * QUETOO_TICK_SECONDS = 60.0
 		return false;
 	}
```

Comment:
Just a couple minor adjustments to `Cl_ValidDeltaEntity` to make it all-encompassing. This makes the conditional that relies on it below simper.